java/clients/src/main/java/sleeper/clients/deploy/properties/GeneratePropertiesDocumentation.java

```diff
@@ -0,0 +1,132 @@
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.function.Consumer;
+
+public class GeneratePropertiesDocumentation {
+
+    private GeneratePropertiesDocumentation() {
+    }
+
+    public static void generateDocumentation(Path root) throws IOException {
+        generateMasterPage(root);
+        Path headPath = Files.createDirectories(root.resolve("docs/usage/properties"));
+
+        //---------- Instance Properties ----------
+        Path instancePath = Files.createDirectories(headPath.resolve("instance/"));
+        InstancePropertyGroup.getAll().forEach(instancePropertyGroup -> {
+            try {
+                writeFile(instancePath.resolve(groupNameToFileName(instancePropertyGroup.getName())),
+                        stream -> writePropertiesMarkdownFile(new InstanceProperties(), instancePropertyGroup, stream));
+            } catch (Exception e) {
+                System.out.println("Unable to write property file for group: " + instancePropertyGroup.getName());
+            }
+        });
+
+        //---------- Table Properties ----------
+        Path tablePath = Files.createDirectories(headPath.resolve("table/"));
+        TablePropertyGroup.getAll().forEach(tablePropertyGroup -> {
+            try {
+                writeFile(tablePath.resolve(groupNameToFileName(tablePropertyGroup.getName())),
+                        stream -> writePropertiesMarkdownFile(new TableProperties(new InstanceProperties()), tablePropertyGroup, stream));
+            } catch (Exception e) {
+                System.out.println("Unable to write property file for group: " + tablePropertyGroup.getName());
+            }
+        });
+    }
+
+    private static void generateMasterPage(Path root) {
+        try {
+            writeFile(root.resolve("docs/usage/property-master.md"),
+                    stream -> generateDocumentLinks(stream));
+        } catch (Exception e) {
+            System.out.println("Unable to generate property master document");
+        }
+    }
+
+    private static void generateDocumentLinks(OutputStream stream) {
+        try {
+            PrintStream out = new PrintStream(stream, true, StandardCharsets.UTF_8.name());
+            out.println("## Instance Properties");
+            out.println();
+            InstancePropertyGroup.getAllGroupNamesAlphabetically().forEach(group -> {
+                out.println(pageLinkFromGroupName(group, "instance/"));
+            });
+            out.println();
+            out.println("## Table Properties");
+            out.println();
+            TablePropertyGroup.getAllGroupNamesAlphabetically().forEach(group -> {
+                out.println(pageLinkFromGroupName(group, "table/"));
+            });
+        } catch (UnsupportedEncodingException e) {
+            System.out.println("Unable to encode output file");
```

Comment:
It looks important to fail the build if an exception happens here, because it means the file didn't write properly. It looks like the exception should be thrown all the way out?